File: app/src/main/java/com/starcut/starflightclient/StarFlightCallback.java
package com.starcut.starflightclient;

public interface StarFlightCallback<T extends StarFlightResponse>
{
    /**
     * Called when the operation succeeds
     * @param result the result of the operation
     */
    public void onSuccess(T result);

    /**
     * Called when the operation fails.
     * @param message error description
     * @param t an exception that occurred, or null if not applicable
     */
    public void onFailure(String message, Throwable t);
}

File: app/src/main/java/com/starcut/starflightclient/starflight/callbacks/StarFlightCallback.kt
package com.starcut.starflightclient.starflight.callbacks

interface StarFlightCallback<T : StarFlightResponse> {
    /**
     * Called when the operation succeeds
     * @param result the result of the operation
     */
    fun onSuccess(result: T)

    /**
     * Called when the operation fails.
     * @param message error description
     * @param t an exception that occurred, or null if not applicable
     */
    fun onFailure(message: String, t: Throwable)
}
